app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -202,8 +202,9 @@ private fun WallpaperThumbnailItem(
     loadWallpaperResource: suspend (Wallpaper) -> Bitmap?,
     isSelected: Boolean,
     aspectRatio: Float = 1.1f,
-    onSelect: (Wallpaper) -> Unit
+    onSelect: suspend (Wallpaper) -> Unit
```

Comment:
The only benefit would be when a child composable had a different lifetime than it's parents, the suspending lambdas could be tied to the most relevant scope. In this case we actually want to avoid that - there's a hypothetical where a wallpaper is selected and then scrolled away which could cancel the download. I'll update as part of this PR

Reply:
To add to that: I think we still want a suspending lambda for `loadWallpaperResource` for two reasons:
- we actually do want that scope tied to the lifetime of the child
- we need a result from it, so can't wrap it in a non-returning `coroutineScope.launch` at the top-level. we would need to refactor the `LoadBitmapUseCase` to return a `Deferred`to do something better here, I think